integration_tests/db_test.go

```diff
@@ -143,4 +146,234 @@ var _ = When("create backstage with CR configured", func() {
+						"rhdh.redhat.com/app": backstageName,
+					},
+					Type: corev1.ServiceTypeClusterIP,
+					Ports: []corev1.ServicePort{
+						{Port: 5432},
+					},
+				},
+			})
+			Expect(err).ShouldNot(HaveOccurred())
+
+			err = k8sClient.Create(ctx, &corev1.Service{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      dbServiceHLName,
+					Namespace: ns,
+				},
+				Spec: corev1.ServiceSpec{
+					Selector: map[string]string{
+						"rhdh.redhat.com/app": backstageName,
+					},
+					ClusterIP: corev1.ClusterIPNone,
+					Ports: []corev1.ServicePort{
+						{Port: 5432},
+					},
+				},
+			})
+			Expect(err).ShouldNot(HaveOccurred())
+
+			err = k8sClient.Create(ctx, &corev1.Secret{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      dbSecretName,
+					Namespace: ns,
+				},
+				StringData: map[string]string{
+					"POSTGRES_HOST":             dbServiceName,
+					"POSTGRES_PORT":             "5432",
+					"POSTGRES_USER":             "postgresql",
+					"POSTGRES_PASSWORD":         "my-awesome-password",       //notsecret
+					"POSTGRESQL_ADMIN_PASSWORD": "my-super-awesome-password", //notsecret
+				},
+			})
+			Expect(err).ShouldNot(HaveOccurred())
+
+			err = k8sClient.Create(ctx, &appsv1.StatefulSet{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      dbStatefulSetName,
+					Namespace: ns,
+				},
+				Spec: appsv1.StatefulSetSpec{
+					Selector: &metav1.LabelSelector{
+						MatchLabels: map[string]string{
+							"rhdh.redhat.com/app": dbStatefulSetName,
+						},
+					},
+					ServiceName: dbServiceHLName,
+					Template: corev1.PodTemplateSpec{
+						ObjectMeta: metav1.ObjectMeta{
+							Labels: map[string]string{
+								"rhdh.redhat.com/app": dbStatefulSetName,
+							},
+							Name: dbStatefulSetName,
+						},
+						Spec: corev1.PodSpec{
+							Containers: []corev1.Container{
+								{
+									Name:  "postgresql",
+									Image: "quay.io/fedora/postgresql-15:latest",
+									EnvFrom: []corev1.EnvFromSource{
+										{
+											SecretRef: &corev1.SecretEnvSource{
+												LocalObjectReference: corev1.LocalObjectReference{
+													Name: dbSecretName,
+												},
+											},
+										},
+									},
+									Ports: []corev1.ContainerPort{
+										{ContainerPort: 5432},
+									},
+								},
+							},
+						},
+					},
+					VolumeClaimTemplates: []corev1.PersistentVolumeClaim{
+						{
+							ObjectMeta: metav1.ObjectMeta{
+								Name: "data",
+							},
+							Spec: corev1.PersistentVolumeClaimSpec{
+								AccessModes: []corev1.PersistentVolumeAccessMode{
+									corev1.ReadWriteOnce,
+								},
+								Resources: corev1.VolumeResourceRequirements{
+									Requests: corev1.ResourceList{
+										corev1.ResourceStorage: resource.MustParse("1Gi"),
+									},
+								},
+							},
+						},
+					},
+				},
+			})
+			Expect(err).ShouldNot(HaveOccurred())
+		})
+
+		It("should reuse existing DB resources when reconciling", func() {
```

Comment:
Not sure what do we test with such a complex procedure...
It looks like that K8s correctly uses object type/name as a key and does not recreate it? Is it the point?

Reply:
Sorry, still a bit confusing to me :) 

It sounds like you want to check 2 things:
1. the names of objects created by operator are expected
2. statefulset "patching" in a case when we try to change immutable fields (serviceName?) works as expected (delete/create)

If that's the case I believe have 2 simpler tests. First as unit test (just compare your model with predictable object names), second as integration test where you create CR and then update it (using raw config apparently) with changed SS. 
Does it make sense?

Reply:
I think the tests showl be as focused as possible. It is still not clear to what does this test check. 

The way I see it is that you try to cover 2 cases: 
- the names of objects we create are expected. I.e:
backstage-<cr-name>,  not <cr-name>-backstage
backstage-psql-<cr-name>,  not backstage-db-<cr-name>
etc.
This one can be covered by simple unit test IMO?
- statefulset "patching" in a case when we try to change immutable fields (serviceName?) works as expected (delete/create), i.e:
the (integration) test can create CR with serviceName expected by 1.1 using raw configuration 
then it can update CR with default serviceName and validate it.
WDYT?

Reply:
Okay, I've tried to change the tests with this trade-off. See [`4758c86` (#369)](https://github.com/janus-idp/operator/pull/369/commits/4758c86e5758f2e294db3c7d8aa0ea31dd6cc8bd)